crates/db_views_moderator/src/modlog_combined_view.rs

```diff
@@ -0,0 +1,1405 @@
+  ExpressionMethods,
+  IntoSql,
+  JoinOnDsl,
+  NullableExpressionMethods,
+  QueryDsl,
+  SelectableHelper,
+};
+use diesel_async::RunQueryDsl;
+use i_love_jesus::PaginatedQueryBuilder;
+use lemmy_db_schema::{
+  aliases,
+  newtypes::{CommentId, CommunityId, PersonId, PostId},
+  schema::{
+    admin_allow_instance,
+    admin_block_instance,
+    admin_purge_comment,
+    admin_purge_community,
+    admin_purge_person,
+    admin_purge_post,
+    comment,
+    community,
+    instance,
+    mod_add,
+    mod_add_community,
+    mod_ban,
+    mod_ban_from_community,
+    mod_feature_post,
+    mod_hide_community,
+    mod_lock_post,
+    mod_remove_comment,
+    mod_remove_community,
+    mod_remove_post,
+    mod_transfer_community,
+    modlog_combined,
+    person,
+    post,
+  },
+  source::combined::modlog::{modlog_combined_keys as key, ModlogCombined},
+  utils::{get_conn, DbPool},
+  InternalToCombinedView,
+  ModlogActionType,
+};
+use lemmy_utils::error::LemmyResult;
+
+impl ModlogCombinedPaginationCursor {
+  // get cursor for page that starts immediately after the given post
+  pub fn after_post(view: &ModlogCombinedView) -> ModlogCombinedPaginationCursor {
+    let (prefix, id) = match view {
+      ModlogCombinedView::AdminAllowInstance(v) => {
+        ("AdminAllowInstance", v.admin_allow_instance.id.0)
+      }
+      ModlogCombinedView::AdminBlockInstance(v) => {
+        ("AdminBlockInstance", v.admin_block_instance.id.0)
+      }
+      ModlogCombinedView::AdminPurgeComment(v) => ("AdminPurgeComment", v.admin_purge_comment.id.0),
+      ModlogCombinedView::AdminPurgeCommunity(v) => {
+        ("AdminPurgeCommunity", v.admin_purge_community.id.0)
+      }
+      ModlogCombinedView::AdminPurgePerson(v) => ("AdminPurgePerson", v.admin_purge_person.id.0),
+      ModlogCombinedView::AdminPurgePost(v) => ("AdminPurgePost", v.admin_purge_post.id.0),
+      ModlogCombinedView::ModAdd(v) => ("ModAdd", v.mod_add.id.0),
+      ModlogCombinedView::ModAddCommunity(v) => ("ModAddCommunity", v.mod_add_community.id.0),
+      ModlogCombinedView::ModBan(v) => ("ModBan", v.mod_ban.id.0),
+      ModlogCombinedView::ModBanFromCommunity(v) => {
+        ("ModBanFromCommunity", v.mod_ban_from_community.id.0)
+      }
+      ModlogCombinedView::ModFeaturePost(v) => ("ModFeaturePost", v.mod_feature_post.id.0),
+      ModlogCombinedView::ModHideCommunity(v) => ("ModHideCommunity", v.mod_hide_community.id.0),
+      ModlogCombinedView::ModLockPost(v) => ("ModLockPost", v.mod_lock_post.id.0),
+      ModlogCombinedView::ModRemoveComment(v) => ("ModRemoveComment", v.mod_remove_comment.id.0),
+      ModlogCombinedView::ModRemoveCommunity(v) => {
+        ("ModRemoveCommunity", v.mod_remove_community.id.0)
+      }
+      ModlogCombinedView::ModRemovePost(v) => ("ModRemovePost", v.mod_remove_post.id.0),
+      ModlogCombinedView::ModTransferCommunity(v) => {
+        ("ModTransferCommunity", v.mod_transfer_community.id.0)
+      }
+    };
```

Comment:
You can use strum to avoid typing out all the enum names. Then you only need to return the id from match.

https://docs.rs/strum/latest/strum/derive.AsRefStr.html